lib/truffle/java/java_utilities.rb

```diff
@@ -0,0 +1,507 @@
+          end
+          if !final_field?(f)
+            mh = unreflect_setter(f)
+            a_proxy.__send__(:define_method,
+                             Foreign.invoke_java_method(FIELD_GET_NAME, f).to_ruby_string + "=") do |v|
+              ::JavaUtilities.wrap_java_value(Foreign.invoke_java_method(mh, java_object, v))
+            end
+          end
+        end
+      end
+    end
+    a_proxy
+  end
+
+  def self.add_instance_methods(a_proxy)
+    methods = Foreign.invoke_java_method(CLASS_GET_DECLARED_METHODS, a_proxy.java_class)
+    # Not using idiomatic Ruby here as we might not have bootstrapped that at this point.
+    methods_size = java_array_size(methods)
+    (0...methods_size).each do |i; m|
+      m = java_array_get(methods, i)
+      if !static_method?(m)
+        mh = unreflect_method(m)
+        if mh == nil && Foreign.invoke_java_method(METHOD_GET_NAME, m).to_ruby_string == "getFields"
+          p "Not adding getFields."
+        end                                                                            
+                                                                          
+        if mh != nil
+          a_proxy.__send__(:define_method,
+                           Foreign.invoke_java_method(METHOD_GET_NAME, m).to_ruby_string) do |*args|
+            args = args.map do |x|
+              ::JavaUtilities.unwrap_java_value(x)
+            end
+            ::JavaUtilities.wrap_java_value(Foreign.invoke_java_method(mh, self.java_object, *args))
+          end
+        end
+      end
+    end
+    a_proxy
+  end
+
+  def self.constant_field?(a_field)
+    modifiers = Foreign.invoke_java_method(FIELD_GET_MODIFIERS, a_field)
+    constant = Modifiers::FINAL | Modifiers::PUBLIC | Modifiers::STATIC
+    constant == modifiers & constant
+  end
+
+  def self.static_field?(a_field)
+    modifiers = Foreign.invoke_java_method(FIELD_GET_MODIFIERS, a_field)
+    modifiers == modifiers | Modifiers::STATIC
+  end
+
+  def self.final_field?(a_field)
+    modifiers = Foreign.invoke_java_method(FIELD_GET_MODIFIERS, a_field)
+    modifiers == modifiers | Modifiers::FINAL
+  end
+
+  def self.static_method?(a_method)
+    modifiers = Foreign.invoke_java_method(METHOD_GET_MODIFIERS, a_method)
+    modifiers == modifiers | Modifiers::STATIC
+  end
+
+  def self.make_object_proxy(a_class)
+    super_class = if a_class == JAVA_OBJECT_CLASS
+                    ConcreteJavaProxy
+                  else
+                    make_proxy(Foreign.invoke_java_method(CLASS_GET_SUPER_CLASS, a_class))
+                  end
+    a_proxy = Class.new(super_class)
+  end
+
+  def self.add_interfaces(a_proxy)
+    interfaces = Foreign.invoke_java_method(CLASS_GET_INTERFACES, a_proxy.java_class)
+    # Not using idiomatic Ruby here as we might not have bootstrapped that at this point.
+    interfaces_size = java_array_size(interfaces)
+    for i in 0...interfaces_size
```

Comment:
`(0...interfaces_size).each do Ii|` or `interfaces_size.times do |i|` (if you wrote this).